From penberg@cs.helsinki.fi  Wed Oct 29 10:35:28 2008
From: Pekka J Enberg <penberg@cs.helsinki.fi>
Date: Tue, 28 Oct 2008 00:20:03 +0200 (EET)
Subject: Staging: w35und: move supported band initialization out of wb35_probe()
To: greg@kroah.com
Cc: pavel@suse.cz, linux-kernel@vger.kernel.org
Message-ID: <Pine.LNX.4.64.0810280019330.9949@melkki.cs.Helsinki.FI>


From: Pekka Enberg <penberg@cs.helsinki.fi>

This patch moves the static struct ieee80211_supported_band initialization out
of w35_probe() because it's really global read-only configuration data.

Signed-off-by: Pekka Enberg <penberg@cs.helsinki.fi>
Acked-by: Pavel Machek <pavel@suse.cz>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/winbond/linux/wbusb.c |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

--- a/drivers/staging/winbond/linux/wbusb.c
+++ b/drivers/staging/winbond/linux/wbusb.c
@@ -32,6 +32,13 @@ static struct ieee80211_channel wbsoft_c
 	{ .center_freq = 2412},
 };
 
+static struct ieee80211_supported_band wbsoft_band_2GHz = {
+	.channels	= wbsoft_channels,
+	.n_channels	= ARRAY_SIZE(wbsoft_channels),
+	.bitrates	= wbsoft_rates,
+	.n_bitrates	= ARRAY_SIZE(wbsoft_rates),
+};
+
 int wbsoft_enabled;
 struct ieee80211_hw *my_dev;
 struct wb35_adapter * my_adapter;
@@ -192,7 +199,6 @@ static int wb35_probe(struct usb_interfa
 	struct usb_device *udev = interface_to_usbdev(intf);
 	struct wbsoft_priv *priv;
 	struct ieee80211_hw *dev;
-	static struct ieee80211_supported_band band;
 	int err;
 
 	usb_get_dev(udev);
@@ -253,12 +259,8 @@ static int wb35_probe(struct usb_interfa
 	dev->channel_change_time = 1000;
 	dev->queues = 1;
 
-	band.channels = wbsoft_channels;
-	band.n_channels = ARRAY_SIZE(wbsoft_channels);
-	band.bitrates = wbsoft_rates;
-	band.n_bitrates = ARRAY_SIZE(wbsoft_rates);
+	dev->wiphy->bands[IEEE80211_BAND_2GHZ] = &wbsoft_band_2GHz;
 
-	dev->wiphy->bands[IEEE80211_BAND_2GHZ] = &band;
 	err = ieee80211_register_hw(dev);
 	if (err)
 		goto error_free_hw;
